### PR TITLE
Adds script to plot genetic vs. geo distance

### PR DESCRIPTION
**Pull request by cassiawag**
As requested in #20, this PR adds a `plot_gen_geo_distance.py` to the folder `analyses`. The script takes `--metadata`, `--alignment`, and `--lat-longs` as inputs and plots genetic distance to closest strain vs. geographic distance to closest strain. This is done for all viruses with `region = seattle` where `location` exists and is a census tract in Washington. The output is `analyses/h3n2_dist_genetic_geo.png`.

### CODE REVIEW

File: analyses/plot_gen_geo_distance.py [hunk 1]
```diff
@@ -3,14 +3,14 @@
 The inputs are --metadata, --alignment, and --lat-longs. The output is a PNG.
 '''
 
+import argparse
```
Comment (trvrb) [inline analyses/plot_gen_geo_distance.py]: Base imports should go before external modules.

File: analyses/plot_gen_geo_distance.py [hunk 2]
```diff
@@ -35,8 +35,8 @@ def strains_mapping(alignment, metadata):
                 if row.strain in diction:
                     diction[row.strain]['location'] = row.location
     mapping = {strains: attributes for strains, attributes in diction.items()
-     if attributes['location'] is not '' and attributes['location'] is not '?'
-     and attributes['location'].startswith('530')}
+               if attributes['location'] != '' and attributes['location'] != '?'
```
Comment (trvrb) [inline analyses/plot_gen_geo_distance.py]: String comparison should be with `==` and `!=`.

File: analyses/plot_gen_geo_distance.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def genetic_distance(mapping):
         for strainB in mapping:
             if strainA != strainB:
                 gen_distance = hamming(mapping[strainA]['seq'], mapping[strainB]['seq'])
-                if gen_distance_low == None or gen_distance < gen_distance_low:
+                if gen_distance_low is None or gen_distance < gen_distance_low:
```
Comment (trvrb) [inline analyses/plot_gen_geo_distance.py]: Comparison to `None` should be with `is` and `is not`.

**Comment by trvrb** [comment]
This is great @cassiawag. My only request is to add `--output-table` and --output-figure` flags as also requested in #25. These can have sensible defaults. But having this will make automated figure export in Snakemake more convenient.

File: analyses/plot_gen_geo_distance.py (hunk 2 above)
Comment (cassiawag) [reply analyses/plot_gen_geo_distance.py]: @trvrb I majorly screwed up this git history. I'm sorry :( Can I just delete this branch and create a new PR?

**Comment by cassiawag** [comment]
@trvrb: I MAJORLY screwed up this git history, and I don't know how to fix it. So I made a new branch called `distance` with the changes you requested and the changes you made to the script and created a PR for that.

**Comment by trvrb** [comment]
@cassiawag: No worries at all. I'll repair Git.

**Comment by trvrb** [comment]
b7939c38b603bab9421759c6779710dd8fe5633f looks messy only because the previous version of `plot_gen_geo_distance.py` in 54aaabbece141c8dda5861827976c915a4a71c8e and previous has a bunch of lines with unnecessary whitespace. This is the big reason I set Atom to strip whitespace.

But this was an easy fix. Things look really good. I'll go ahead and merge this.

BTW your strategy of making a new clean branch is good one. It's always a safe way to deal with Git issues.

**Comment by trvrb** [comment]
I decided to merge in #26 instead. Closing this PR unmerged.

**Pull request closed without merging**